### PR TITLE
Battery Diagnostics for Segbot_v3

### PR DESCRIPTION
**Pull request by maxsvetlik**
These changes make the battery diagnostics node (voltage monitor and emailer) more flexible, allowing compatibility with both segbot_v2 and segbot_v3. These changes include:

- Adding additional parameters into the segbot_v3.launch file, namely the voltage cutoff for a discharged battery

- Modifying segbot_v3_controller to publish voltage values on the /battery0 topic, which is expected by the battery monitor

### CODE REVIEW

File: segbot_bringup/scripts/segbot_v3_controller [hunk 1]
```diff
@@ -10,6 +10,8 @@ from geometry_msgs.msg import Quaternion, Twist
 from nav_msgs.msg import Odometry
 from segway_msgs.msg import ConfigCmd
 from segway_msgs.msg import Status
+from segway_msgs.msg import AuxPower
+from smart_battery_msgs.msg import SmartBatteryStatus
```
Comment (piyushk) [inline segbot_bringup/scripts/segbot_v3_controller]: package.xml will need an exec_depend/run_depend on smart_battery_msgs

File: segbot_sensors/src/segbot_sensors/battery_diagnostics.cpp [hunk 2]
```diff
@@ -41,7 +48,7 @@ int sendMail(std::string outboundList, std::string sender){
   if(mailpipe != NULL){
     std::string to = "To: " + outboundList + "\n";
     std::string from = "From: " + sender + "\n";
-    std::string subject = "Subject: Segbot Altert: " + lhn + " Has a Low Battery\n";
+    std::string subject = "Subject: Segbot Alert: " + lhn + " Has a Low Battery\n";
```
Comment (piyushk) [inline segbot_sensors/src/segbot_sensors/battery_diagnostics.cpp]: I'm gonna miss this typo.

File: segbot_sensors/src/segbot_sensors/battery_diagnostics.cpp [hunk 3]
```diff
@@ -71,27 +90,25 @@ int main(int argc, char **argv){
...
   status_val.key = "battery_value";
-  status_val.value = "183.0"; //dummy value
+  status_val.value = "183.0"; 			//dummy value
   std::ostringstream ss;
 
-  //parameters
-  n.param("weighted_average_const", alpha, .25);
-
+  
   while (ros::ok()){
```
Comment (piyushk) [inline segbot_sensors/src/segbot_sensors/battery_diagnostics.cpp]: most of our other code has spaces between ')' and '{', and 'if'/'for' and '('

**Comment by piyushk** [comment]
I've added a couple of minor comments. Is there a process to setting up the sendmail stuff on a computer? Does it need to be documented on the wiki?

**Comment by maxsvetlik** [comment]
I addressed your comments with bb1cca6. Let me know if you have others

The sendmail setup script is located at: segbot_sensors/sendmail_configure.sh.

It pretty much takes care of everything. The script only needs to be run once and is written such that the external email & password are not stored in plaintext. Consequently, the one running the script will need to enter the external email & password during the sendmail setup process. 

The password shouldn't be documented on the wiki (obviously) but I will add some instructions regarding the battery monitor & emailer.

**Comment by piyushk** [comment]
Thanks Max!

@jack-oquin This commit adds the battery monitor to v3. As such, I believe the v3 is now up-to the v2 functionality.

**Comment by jack-oquin** [comment]
Looks good to me.

There is a new [sensor_msgs/BatteryState](http://docs.ros.org/indigo/api/sensor_msgs/html/msg/BatteryState.html) message we should probably migrate to at some point, but that's a separate issue.

**Comment by jack-oquin** [comment]
I like the way Max was able to combine the battery monitoring and launch scripts for the two models.

Well done!

**Pull request merged**